packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -75,31 +76,108 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
         port = getInt("port")
     }
 
-    val type =
-        getString("type")?.let {
-            when (it.lowercase(Locale.US)) {
+    val type = getString("type")
+
+    if (address != null && port != null && type != null) {
+        return JSONProxyConfiguration(
+            type,
+            address,
+            port,
+            getString("username"),
+            getString("password")
+        ).asProxyConfig()
+    } else {
+        return null
+    }
+}
+
+@Suppress("LabeledExpression")
+internal fun ReadableArray.asFirstPartyHosts(): Map<String, Set<TracingHeaderType>> {
+    val firstPartyHosts = this.toArrayList() as List<ReadableMap>
+    return firstPartyHosts.mapNotNull<ReadableMap, JSONFirstPartyHost> {
+        it.getString("match")?.let { match ->
```

Comment:
seems it can be written it a bit easier way:

```
val match = it.getString("match")
val propagatorTypes = it.getArray("propagatorTypes")
if (match != null && propagatorTypes != null) {
   JSONFirstPartyHost(match, propagatorTypes.toArrayList() as List<String>)
} else null
```

imo easier to read